### ISSUE TITLE
allow user to update avatar

### ISSUE THREAD

**Issue by pdpinch** [mitodl/micromasters#179]
Allow users to upload an image, to replace their edX avatar

![image](https://cloud.githubusercontent.com/assets/430126/18056441/e0bb2eb0-6dd9-11e6-9564-a0f639888fb1.png)

![image](https://cloud.githubusercontent.com/assets/430126/18098882/2ac6de86-6eb2-11e6-8ce2-39b25dbe932b.png)

![image](https://cloud.githubusercontent.com/assets/430126/18098901/3b63cdc6-6eb2-11e6-9ad0-b431b4ba14e9.png)

![image](https://cloud.githubusercontent.com/assets/430126/18098954/6f3bd10c-6eb2-11e6-8c2f-63e4c063c730.png)

# Acceptance Criteria

- [ ] Display current avatar
- [ ] On clicking, open a model 
- [ ] allow user to drag an image or select a file to upload
- [ ] After upload is complete, allow user to crop image
- [ ] Save image using an API (#394)
      - [ ] Use django-storages back-end to manage location for saving
      - [ ] Same authorizations as profile API

**Comment by pdpinch** [comment]
@noisecapella before you and @gsidebo work on this, I'd like to investigate using the gravatar service.

**Comment by giocalitri** [comment]
this depends on #394

**Comment by giocalitri** [comment]
we are going to postpone this story

**Comment by roberthouse54** [comment]
I feel that gravatar use is pretty low, especially globally, and think we should implement our own UI for uploading and saving user images. I have posted basic designs Here:

https://invis.io/KT85HTW9V
https://invis.io/MB85HUX3S

**Comment by pdpinch** [comment]
Is there a screenshot of the state before the file dialog comes up (i.e. before https://invis.io/KT85HTW9V)

I don't suppose there's a component for doing image editing like that?

**Comment by roberthouse54** [comment]
I don't know if there is a component for cropping. I was trying to keep the UI pretty simple, so I was thinking that clicking the "Edit Photo" link just directly opens the file browser. (for V1). Other sites have a dialog with other options....

**Comment by pdpinch** [comment]
FYI @roberthouse54 I moved this near the top of the backlog. Ecommerce issues will go above it, but not additional search facets.

**Comment by pdpinch** [comment]
@roberthouse54 what do you think of http://dropsofserenity.github.io/react-avatar-cropper/ ?

We'd have to see if we can add a mask to make the image a circle, but otherwise it seems to have the necessary functionality. 

React code available from npm: https://www.npmjs.com/package/react-avatar-cropper

**Comment by roberthouse54** [comment]
This looks like a great solution to me. The mask should not be a problem, as it is just a border radius on the element.

**Comment by aliceriot** [comment]
This looks a little closer to what's in the mockups: https://github.com/DominicTobias/react-image-crop

**Comment by roberthouse54** [comment]
Either one is OK with me. We need to be cropping at a fixed (square) aspect ratio, which I see this one does.

**Comment by aliceriot** [comment]
another two options: 

https://github.com/roadmanfong/react-cropper
https://github.com/instructure-react/react-crop

**Comment by aliceriot** [comment]
Ok, I think react-cropper looks like the most flexible, it's a React wrapper around a more mature JS library (https://github.com/fengyuanchen/cropperjs)

**Comment by aliceriot** [comment]
I can do a fixed aspect-ratio as well

**Comment by amir-qayyum-khan** [comment]
Update gravatar with latest image Rob provided me. So that profile image URL(s) get update in api ``/api/v0/profiles/staff/``

![avatar_default](https://cloud.githubusercontent.com/assets/10431250/18322315/85b71d34-754b-11e6-8245-ccf696b7e459.png)

cc @roberthouse54 @pdpinch
